scipy/spatial/transform/tests/test_rotation.py

```diff
@@ -889,3 +889,117 @@ def test_slerp_call_time_out_of_range():
         s([0, 1, 2])
     with pytest.raises(ValueError, match="times must be within the range"):
         s([1, 2, 6])
+
+
+def test_spline_trivial():
+    np.random.seed(0)
+
+    key_rots = Rotation.random(10)
+    key_times = np.arange(10)
+
+    spline = QSpline(key_times, key_rots)
+    assert_allclose(spline(key_times).as_rotvec(), key_rots.as_rotvec())
+
+
+def test_spline_c1_continuity():
+    # Angular velocity continuity satisfied for smooth changes in orientation
+    key_rots = Rotation.from_euler('z', [0, 10, 20, 30, 20, 10, 0],
+                                   degrees=True)
+    key_times = np.arange(7)
+
+    spline = QSpline(key_times, key_rots,)
+    test_times = key_times[1:-1]
+    dt = 1e-7
+
+    assert_allclose(
+        spline.angular_velocity(test_times + dt),
+        spline.omega[1:-1], atol=1e-6, rtol=1e-6
+    )
+
+    assert_allclose(
+        spline.angular_velocity(test_times - dt),
+        spline.omega[1:-1], atol=1e-6, rtol=1e-6
+    )
+
+
+def test_spline_constant_angular_velocity():
+    key_rots = Rotation.from_euler('z', [10, 20, 30], degrees=True)
+    key_times = [0, 1, 2]
+
+    omega_i = [0, 0, 10 * np.pi/180]
+    omega_f = [0, 0, 10 * np.pi/180]
+    spline = QSpline(key_times, key_rots, omega_i=omega_i, omega_f=omega_f)
+
+    vel = spline.omega[1]
+    assert_allclose(vel, omega_i)
+    assert_allclose(vel, omega_f)
```

Comment:
`omega_i` and `omega_f` are identical list objects in this unit test; why the assertion for `vel` compared to both as apposed to a single reference value? Are you trying to ensure that `QSpline` doesn't modify the omega argument objects passed in?

Reply:
You're right. Only one check is required here.